target/linux/ar71xx/base-files/lib/upgrade/platform.sh

```diff
@@ -589,6 +588,21 @@ platform_check_image() {
 		tplink_pharos_check_image "$1" "01000000" "$(tplink_pharos_v2_get_model_string)" '\0\xff\r' && return 0
 		return 1
 		;;
+        cpe510)
+               tplink_v2_check() {
+                        local part
+                        part=$(find_mtd_part 'product-info')
+                        dd if=$part bs=1 skip=4360 count=64 2>/dev/null | tr -d '\r\0' | head -n 1
+                        }
+                        md_string="$(tplink_v2_check)"
+                        if [ "${md_string:35:3}" == "2.0" ] || [ "${md_string:26:3}" == "2.0" ] ; then
+				tplink_pharos_check_image "$1" "7f454c46" "$(tplink_pharos_v2_get_model_string)" '\0\xff\r' && return 0
```

Comment:
Like above:
- Inline function declaration
- CPE510v1 compatibility
- reuse md_string

Reply:
Sure about the "7f454c46" in contrast to the "01000000" on the CPE210v2?